plugins/org.jkiss.dbeaver.ext.tidb/src/org/jkiss/dbeaver/ext/tidb/mysql/model/TiDBMySQLDataSource.java

```diff
@@ -145,4 +152,19 @@ public boolean isServerVersionAtLeast(int major, int minor) {
         }
         return true;
     }
+    
+    @Override
+    protected Map<String, String> getInternalConnectionProperties(DBRProgressMonitor monitor, DBPDriver driver, JDBCExecutionContext context, String purpose, DBPConnectionConfiguration connectionInfo) throws DBCException {
+        Map<String, String> props = super.getInternalConnectionProperties(monitor, driver, context, purpose, connectionInfo);
+        // build application name
+        String appName = DBUtils.getClientApplicationName(getContainer(), context, purpose);
+        appName = "dbeaver_tidb_plugin" + (appName.isEmpty() ? "" : "(" + appName + ")");
+        
+        // build conAttr value
+        String connAttr = props.get(CONN_ATTR_NAME);
+        connAttr = PROP_APPLICATION_NAME + ":" + appName + (connAttr == null || connAttr.isEmpty() ? "" : "," + connAttr);
```

Comment:
```suggestion
        connAttr = PROP_APPLICATION_NAME + ":" + appName + (CommonUtils.isEmpty(connAttr) ? "" : "," + connAttr);
```

+add CommonUtils import

Reply:
Added.